packages/slate-editor/src/extensions/placeholders/PlaceholdersExtension.tsx

```diff
@@ -80,7 +80,6 @@ export function PlaceholdersExtension({
             return undefined;
         },
         isRichBlock: PlaceholderNode.isPlaceholderNode,
-        isVoid: PlaceholderNode.isPlaceholderNode,
```

Comment:
@e1himself this is the reason why pasting didn't work - looks like slate detects isVoid element to disallow pasting. Placeholders are not void, they have inputs. What do you think?

Reply:
@aspirisen this doesn't sound right. 

I believe void vs not void is telling Slate how element `children` are handled. 

https://docs.slatejs.org/api/nodes/element#void-vs-not-void

> ### Void vs Not Void
> 
> In a not "void" element, Slate handles the rendering of its children (e.g. in a paragraph where the Text and Inline children are rendered by Slate). In a "void" element, the children are rendered by the Element's render code.

With this definition, Placeholder is a void node. As its `children` array doesn't mean anything and can only be empty (i.e. `[{ text: "" }]`).

Reply:
hm, but the placeholder allows user input, we just render it as an input element, but we could render it as a `contenteditable` div that looks like the input and ux would be the same

I thought about void elements like about elements that does not imply user input, like images, dividers

Reply:
But you would lose all the capabilities of `input` element, no? Like validation, pattern, focus etc.

Reply:
I think it is implementation details, placeholders are part of slate ui and support text input

@e1himself what will happen if placeholders will be non-void elements?

Reply:
This is tricky one, one the one hand we mark placeholder as an void element, but on the other slate for some reason forbids pasting into void elements

Reply:
> hm, but the placeholder allows user input, we just render it as an input element, but we could render it as a contenteditable div that looks like the input and ux would be the same

No, please. This is clearly not the right path.

> but on the other slate for some reason forbids pasting into void elements

This doesn't fit into my understanding of how Slate works.

Let's try to find the exact line of code that forbids it? Maybe this is a bug and we'll be able to report and fix it. Or maybe it's something else.

Reply:
> hm, but the placeholder allows user input, we just render it as an input element, but we could render it as a contenteditable div that looks like the input and ux would be the same

No, please. This is clearly not the right path.

> but on the other slate for some reason forbids pasting into void elements

This doesn't fit into my understanding of how Slate works.

Let's try to find the exact line of code that forbids it? Maybe this is a bug and we'll be able to report and fix it. Or maybe it's something else on our side.

Reply:
Created an issue in slate https://github.com/ianstormtaylor/slate/issues/5294